src/axom/quest/SamplingShaper.hpp

```diff
@@ -456,6 +456,53 @@ class SamplingShaper : public Shaper
   //@}
 
 public:
+  /**
+   * \brief Import an initial set of material volume fractions before shaping
+   *
+   * \param [in] initialGridFuncions The input data as a map from material names to grid functions
+   * 
+   * The imported grid functions are interpolated at quadrature points and registered
+   * with the supplied names as material-based quadrature fields
+   */
+  void projectInitialVolumeFractions(
+    const std::map<std::string, mfem::GridFunction*> initialGridFunctions)
+  {
+    internal::ScopedLogLevelChanger logLevelChanger(
+      this->isVerbose() ? slic::message::Debug : slic::message::Warning);
+
+    auto* mesh = m_dc->GetMesh();
+
+    // ensure we have a starting quadrature field for the positions
+    if(!m_inoutShapeQFuncs.Has("positions"))
+    {
+      shaping::generatePositionsQFunction(mesh,
+                                          m_inoutShapeQFuncs,
+                                          m_quadratureOrder);
+    }
+    auto* positionsQSpace = m_inoutShapeQFuncs.Get("positions")->GetSpace();
+
+    // Interpolate grid functions at quadrature points & register material quad functions
+    // assume all elements have same integration rule
+    for(auto& entry : initialGridFunctions)
+    {
+      const auto& name = entry.first;
+      auto* gf = entry.second;
+
+      if(gf == nullptr)
```

Comment:
Is there a case where a named map entry with a `nullptr` grid function is a valid case, or should we add a warning for the user?

Reply:
It's not clear yet... I can imagine passing in all the initial materials w/ some of them being nullptrs.

It's easy enough to filter these out on the user side, so it's probably a good idea to add warnings.

Reply:
done